pilot/pkg/serviceregistry/kube/controller/ambient/statusqueue/conversion.go

```diff
@@ -42,32 +41,40 @@ type conditions struct {
 // translateToPatch converts a ConditionSet to a patch
 // The ConditionSet is expected to contain all Types owned by this controller; if they should be unset, they should nil.
 // Failure to do so means the status may not properly get pruned.
-func translateToPatch(object model.TypedObject, status model.ConditionSet, currentConditionsList []string) []byte {
-	currentConditions := smallset.New(currentConditionsList...)
+// currentConditions must be set to the current conditions. This is used to determine if we need a write.
+// Note: we do not simply rely on SSA to tell us if a change is needed, because every write does a mutation due to LastTransitionTime.
+func translateToPatch(object model.TypedObject, status model.ConditionSet, currentConditions map[string]model.Condition) []byte {
 	conds := make([]any, 0, len(status))
 
-	needsDelete := false
+	changesNeeded := false
 
-	for t, values := range status {
-		if currentConditions.Contains(string(t)) {
-			needsDelete = true
+	for t, v := range status {
+		nowExists := v != nil
+		cc, curExists := currentConditions[string(t)]
```

Comment:
Took me a sec to figure out the difference between now and cur. Naming is hard. This might be slightly better:
```suggestion
		shouldExist := v != nil
		cc, curExists := currentConditions[string(t)]
```